aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -761,6 +769,42 @@ module aptos_framework::staking_contract {
         distribute_internal(staker, operator, staking_contract, &mut store.distribute_events);
     }
 
+    /// Destroy an owned staking contract and take ownership of its underlying stake pool.
+    /// Extracted state and capabilities will be used to construct a delegation pool of same owner, operator
+    /// and commission fee.
+    public(friend) fun destroy_staking_contract(
+        staker: &signer,
+        operator: address,
+    ): (u64, address, OwnerCapability, u64, Pool, SignerCapability) acquires Store, BeneficiaryForOperator {
+        let staker_address = signer::address_of(staker);
+        assert_staking_contract_exists(staker_address, operator);
+
+        let store = borrow_global_mut<Store>(staker_address);
+        let (_, staking_contract) = simple_map::remove(&mut store.staking_contracts, &operator);
+
+        // withdraw entire inactive stake and unlock commission produced so far
+        distribute_internal(staker_address, operator, &mut staking_contract, &mut store.distribute_events);
+        request_commission_internal(
```

Comment:
Can we add a comment here that any pending inactive from commission distribution will be handled as part of the conversion in delegation_pool and converted to shares in the new delegation pool's pending inactive?